### PR TITLE
Solaris updates

### PR DESCRIPTION
**Pull request by cgzones**
Tested on OmniOS

### CODE REVIEW

File: solaris/Platform.c [hunk 1]
```diff
@@ -255,7 +259,7 @@ static int Platform_buildenv(void* accum, struct ps_prochandle* Phandle, uintptr
       return 1;
    }
    strlcpy( accump->env + accump->size, str, (accump->capacity - accump->size));
-   strncpy( accump->env + accump->size + thissz + 1, "\n", 1);
+   strncpy( accump->env + accump->size + thissz + 1, "\n", 2);
```
Comment (BenBE) [inline solaris/Platform.c:262]: Is the ` + 1` in this line correct? Because the `strlcpy` only should copy `thissz` bytes into the buffer …

**Review by BenBE — COMMENTED** [review]
One minor remark/need for clarification. Apart from this LGTM.

File: solaris/Platform.c (hunk 1 above)
Comment (cgzones) [reply solaris/Platform.c:262]: It worked&#8482;, it's also `strncpy`.
Sadly valgrind and address sanitizer were not available.

**Comment by RussianE39** [comment]
Not buildable on Solaris 11.4 SRU32 / SPARC. Complete configure/buildlog attached.
[htop-pr633_build_log.txt](https://github.com/htop-dev/htop/files/6513976/htop-pr633_build_log.txt)

File: solaris/SolarisProcessList.h [hunk 2]
```diff
@@ -13,6 +13,9 @@ in the source distribution for its full text.
 #define GZONE "global    "
 #define UZONE "unknown   "
 
+
+#include "config.h" // IWYU pragma: keep
+
 #include <kstat.h>
```
Comment (BenBE) [inline solaris/SolarisProcessList.h:19]: Missing `stdbool.h`, `stdint.h`.

**Comment by BenBE** [comment]
> Not buildable on Solaris 11.4 SRU32 / SPARC.

Thank you for the report. What commit did you use to compile? From `htop-dev/master` or did you try `cgzones/solaris`?

**Comment by RussianE39** [comment]
> > Not buildable on Solaris 11.4 SRU32 / SPARC.
> 
> Thank you for the report. What commit did you use to compile? From `htop-dev/master` or did you try `cgzones/solaris`?

I've used cgzones/solaris. htop-dev/master not buildable on solaris either.

**Comment by BenBE** [comment]
> > > Not buildable on Solaris 11.4 SRU32 / SPARC.
> > 
> > 
> > Thank you for the report. What commit did you use to compile? From `htop-dev/master` or did you try `cgzones/solaris`?
> 
> I've used cgzones/solaris. htop-dev/master not buildable on solaris either.

Strange. Can you check that `solaris/SolarisProcessList.h` in line 27 references `#include "ProcessList.h"` and that declaring some variable `extern ProcessList foo;` right below that line works?

You might also want to add the two includes I mentioned in the review comment above, but those shouldn't affect the `ProcessList` compilation issue.

File: solaris/SolarisProcessList.h [hunk 3]
```diff
@@ -21,6 +24,10 @@ in the source distribution for its full text.
 #include <sys/sysinfo.h>
 #include <sys/swap.h>
 
+#include "ProcessList.h"
```
[the remark this answers is not in the document]
Comment (RussianE39) [reply solaris/SolarisProcessList.h:27]: In file included from solaris/SolarisProcessList.c:10:
./solaris/SolarisProcessList.h:53:56: error: unknown type name ‘SolarisProcess’; did you mean ‘SolarisProcessList’?
   53 | char* SolarisProcessList_readZoneName(kstat_ctl_t* kd, SolarisProcess* sproc);
      |                                                        ^~~~~~~~~~~~~~
      |                                                        SolarisProcessList
./solaris/SolarisProcessList.h:65:33: error: unknown type name ‘psinfo_t’; did you mean ‘vminfo_t’?
   65 | int SolarisProcessList_walkproc(psinfo_t* _psinfo, lwpsinfo_t* _lwpsinfo, void* listptr);
      |                                 ^~~~~~~~
      |                                 vminfo_t
./solaris/SolarisProcessList.h:65:52: error: unknown type name ‘lwpsinfo_t’; did you mean ‘sysinfo_t’?
   65 | int SolarisProcessList_walkproc(psinfo_t* _psinfo, lwpsinfo_t* _lwpsinfo, void* listptr);
      |                                                    ^~~~~~~~~~
      |                                                    sysinfo_t
solaris/SolarisProcessList.c:25:10: fatal error: solaris/ProcessList.h: No such file or directory

**Comment by cgzones** [comment]
> Not buildable on Solaris 11.4 SRU32 / SPARC. Complete configure/buildlog attached.
> [htop-pr633_build_log.txt](https://github.com/htop-dev/htop/files/6513976/htop-pr633_build_log.txt)

containing

> ```
> solaris/Platform.c: In function Platform_getMaxPid:
> solaris/Platform.c:166:44: warning: passing argument 2 of kstat_lookup discards const qualifier from pointer target type [-Wdiscarded-qualifiers]
>   166 |       kstat_t* kshandle = kstat_lookup(kc, "unix", 0, "var");
>       |
> ```

is definitely not this this pull request branch, as the `kstat_lookup` calls have been wrapped.

**Comment by RussianE39** [comment]
Just checked htop-dev master - builds and works under S11.4/SPARC now. Thanks

**Comment by cgzones** [comment]
@RussianE39 many thanks for testing in the first place and for verifying.

**Pull request merged**